code/datums/components/material_container.dm

```diff
@@ -73,17 +76,17 @@
 	var/datum/callback/pc = precondition
 	if(pc && !pc.Invoke(user))
 		return
-	var/material_amount = get_item_material_amount(I)
+	var/material_amount = get_item_material_amount(I, breakdown_flags)
 	if(!material_amount)
 		to_chat(user, "<span class='warning'>[I] does not contain sufficient materials to be accepted by [parent].</span>")
 		return
 	if(!has_space(material_amount))
 		to_chat(user, "<span class='warning'>[parent] is full. Please remove materials from [parent] in order to insert more.</span>")
 		return
-	user_insert(I, user)
+	user_insert(I, user, breakdown_flags)
 
 /// Proc used for when player inserts materials
-/datum/component/material_container/proc/user_insert(obj/item/I, mob/living/user)
+/datum/component/material_container/proc/user_insert(obj/item/I, mob/living/user, breakdown_flags=src.breakdown_flags)
```

Comment:
Why the src.breakdown_flags? is that needed to achieve this?

Reply:
I put that there so that if this is called from elsewhere, such as the remote materials component, and this isn't overridden it will default to breaking down the materials through its own methods. I could move it into the proc body if you want me to.